src/imitation/rewards/reward_function.py

```diff
@@ -0,0 +1,35 @@
+"""Type alias shared by reward-related code."""
+
+from typing import Protocol
+import abc
+import numpy as np
+
+class RewardFn(Protocol):
+    """Abstract class for reward function.
+
+    Requires implementation of __call__() to compute the reward given a batch of
+    states, actions, next states and dones.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    @abc.abstractmethod
+    def __call__(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+    ) -> np.ndarray:
+        """Compute rewards for a batch of transitions.
+
+        Args:
+            state: Current states of shape `(batch_size,) + state_shape`.
+            action: Actions of shape `(batch_size,) + action_shape`.
+            next_state: Successor states of shape `(batch_size,) + state_shape`.
+            done: End-of-episode (terminal state) indicator of shape `(batch_size,)`.
+        Returns:
+            Computed np.ndarray rewards of shape `(batch_size,`).
```

Comment:
```suggestion
            Computed rewards of shape `(batch_size,`).
```
Type already documented in type annotation. Could keep it in docstring if you think it's worth drawing attention to, but otherwise I'd lean towards omitting it. WDYT?